alf/networks/transformer_networks.py

```diff
@@ -222,3 +227,167 @@ def forward(self, inputs, state=()):
+                the input. If nested, then ``preprocessing_combiner`` must not be
+                None.
+            input_preprocessors (nested InputPreprocessor): a nest of
+                ``InputPreprocessor``, each of which will be applied to the
+                corresponding input. If not None, then it must have the same
+                structure with ``input_tensor_spec``. This arg is helpful if you
+                want to have separate preprocessings for different inputs by
+                configuring a gin file without changing the code. For example,
+                embedding a discrete input before concatenating it to another
+                continuous vector.
+            preprocessing_combiner (NestCombiner): preprocessing called on
+                complex inputs. Note that this combiner must also accept
+                ``input_tensor_spec`` as the input to compute the processed
+                tensor spec. For example, see ``alf.nest.utils.NestConcat``. This
+                arg is helpful if you want to combine inputs by configuring a
+                gin file without changing the code.
+            fc_layer_params (tuple[int]): a tuple of integers
+                representing FC layer sizes for generating embeddings.
+            activation (nn.functional): activation used for all the layers but
+                the last layer.
+            kernel_initializer (Callable): initializer for all the layers but
+                the last layer. If None, a variance_scaling_initializer will be
+                used.
+            use_fc_bn (bool): whether use Batch Normalization for fc layers.
+            num_of_heads (int): number of heads for the mult-head attention
+            last_layer_size (None): nt used; for interface compatibility
+            last_activation (None): not used; for interface compatibility
+            last_kernel_initializer (None): not used; for interface compatibility
+            last_use_fc_bn (None): not used; for interface compatibility
+            name (str):
+        """
+        super().__init__(
+            input_tensor_spec,
+            input_preprocessors,
+            preprocessing_combiner=preprocessing_combiner,
+            name=name)
+
+        if kernel_initializer is None:
+            kernel_initializer = functools.partial(
+                variance_scaling_init,
+                mode='fan_in',
+                distribution='truncated_normal',
+                nonlinearity=activation)
+
+        embedding_layers = nn.ModuleList()
+        print(self._processed_input_tensor_spec)
+        assert self._processed_input_tensor_spec.ndim == 2, (
+            "expect the "
+            "processed spec to have the shape of [entity_num, feature_dim]")
+
+        for size in fc_layer_params:
+            embedding_layers.append(
+                layers.FC(
+                    self._processed_input_tensor_spec.shape[-1],
```

Comment:
the input size will not correct if there are multiple fc_layer_params

Reply:
You are right. This is a bug. Fixed.